data/common.yaml

```diff
@@ -0,0 +1,13 @@
+---
+profile_mysql_server::create_mysql_home: true
+profile_mysql_server::dbs: {}
+profile_mysql_server::mysql_gid: 238
+profile_mysql_server::mysql_groupname: "mysql"
+profile_mysql_server::mysql_home: "/var/lib/mysql"
+profile_mysql_server::mysql_logdir: "/var/log/mariadb"
+profile_mysql_server::mysql_parentdir: null
+profile_mysql_server::mysql_rundir: "/var/run/mariadb"
+profile_mysql_server::mysql_uid: 238
```

Comment:
Where does this uid/gid come from? Is this default for the mysql/mariadb rpm install?

Reply:
I'm not sure now. I _thought_ it was kind of standard but now that I look into it I don't see that's the case.

The mariadb-server v10.3 RPMs from Red Hat seem to set the IDs to 27 (if the user/group don't exist).

The Mariadb-server 10.5 RPM from MariaDB just lets the system pick (if finds they don't exist and it adds them).

Do you think we should make 27 be default, or stick with 238 as an in-house default? Or make them optional (in which case I think we have to not manage any of the folders if those aren't specified, because if we don't specify what IDs to use Puppet will default to...probably root:root)?

Reply:
Maybe we only set the IDs if we need to. We only need to set the IDs for stateless nodes, correct?

So, unless a node is stateless, we do not set the ID.

Or perhaps even better, could we look at a stateless node's persistent data and figure out what the UID/GID was historically?

Reply:
Discussed a bit in Slack, but yep, pretty much this is for stateless nodes w/ stateful storage. (Or I guess stateful nodes that might be reinstalled but maintain DBs on separate storage.) And particularly where the RPM installer might not reinstall the user/group with consistent IDs.

I've updated so that $mysql_gid and $mysql_uid are optional. If both are provided in Hiera, then the user, group, and associated folders are managed. If neither is defined then this is all skipped. (If only one is defined then Puppet fails with an appropriate message.)